docs/reference/search/search-fields.asciidoc

```diff
@@ -4,33 +4,203 @@
 
 By default, each hit in the search response includes the document
 <<mapping-source-field,`_source`>>, which is the entire JSON object that was
-provided when indexing the document. If you only need certain source fields in
-the search response, you can use the <<source-filtering,source filtering>> to
-restrict what parts of the source are returned.
+provided when indexing the document. To retrieve specific fields in the search
+response, you can use the `fields` parameter:
 
-Returning fields using only the document source has some limitations:
+[source,console]
+----
+GET /_search
+{
+  "query": {
+    "term": {
+      "user.id": "8a4f500d"
+    }
+  },
+  "fields": ["user.name", "timestamp"],
+  "_source": false
+}
+----
 
-* The `_source` field does not include <<multi-fields, multi-fields>> or
-<<alias, field aliases>>. Likewise, a field in the source does not contain
-values copied using the <<copy-to,`copy_to`>> mapping parameter.
-* Since the `_source` is stored as a single field in Lucene, the whole source
-object must be loaded and parsed, even if only a small number of fields are
-needed.
+The `fields` parameter consults both a document's `_source` and the index
+mappings to load and return values. Because it makes use of the mappings,
+`fields` has some advantages over referencing the `_source` directly: it
+accepts <<multi-fields, multi-fields>> and <<alias, field aliases>>, and
+also formats field values like dates in a consistent way.
 
-To avoid these limitations, you can:
+A document's `_source` is stored as a single field in Lucene. So the whole
```

Comment:
It is a little sad your API can't fix this in a cleaner way too. Like, if we only had metadata about the `_source` we could think about it. But, I think, for now this does stay.

Reply:
> It is a little sad your API can't fix this in a cleaner way too.

I also think we should address this, but that it should be a separate change (instead of tackling it in this first cut of the feature).